src/app/commons/models/permissions-model.js

```diff
@@ -0,0 +1,216 @@
+angular.module('wegas.models.permissions', [])
+    .service('PermissionsModel', function($http, $q, $interval, Auth, Responses) {
+        var model = this;
+        model.getSessionPermissions = function(session) {
+            var deferred = $q.defer();
+            $http.get(ServiceURL + "rest/Extended/User/FindAccountPermissionByInstance/g" + session.id).success(function(data) {
+                permissionsToReturn = [];
+                _(data).each(function(account, i) {
+                    var permissions = [],
+                        pattern = new RegExp("^Game:(.*):g" + session.id + "$");
+
+                    // For each permission of each account...
+                    _(account.permissions).each(function(permission, j) {
+                        // Is permission linked with current game ?
+                        if (pattern.test(permission.value)) {
+                            var localPermission = permission.value.match(pattern)[1].split(",");
+                            permissions = permissions.merge(localPermission)
+                        }
+                    });
+                    if (permissions.indexOf("View") >= 0 && permissions.indexOf("Edit") >= 0) {
+                        permissionsToReturn.push(account);
+                    }
+                });
+                deferred.resolve(permissionsToReturn);
+            }).error(function(data) {
+                deferred.resolve(false);
+            });
+            return deferred.promise;
+        };
+
+        /* Add a new trainer to the session */
+        model.addSessionPermission = function(session, trainers, trainer) {
```

Comment:
Seems that `trainer` param is more a `trainerId` like in `removeSessionPermission`